tests/postgres_tests/test_array.py

```diff
@@ -737,6 +737,14 @@ def test_already_converted_value(self):
         vals = ['a', 'b', 'c']
         self.assertEqual(field.clean(vals), vals)
 
+    def test_has_changed(self):
+        field = SimpleArrayField(forms.CharField())
+        self.assertFalse(field.has_changed(None, None))
+        self.assertFalse(field.has_changed(None, ''))
+        self.assertFalse(field.has_changed(None, []))
```

Comment:
Are [], None, and '' all realistic values for `data`? Generally I would expect only string values from POST data.

Reply:
for an empty field `''` is passed in POST data.
if you are referring to pure html form implementation, then `None` and `[]` are not encountered. but if the form is used otherwise (validating api requests) you might expect to see these values.